docs/doc_3_migration.md

```diff
@@ -0,0 +1,87 @@
+# Atlas-SDK-GO Migration Guide
```

Comment:
Note that this content have been reviewed already in google doc. 
Moving it here and adjusting some links. 
I would expect more changes as we keep introduce some fixes.